scripts/apply-ksappend.py

```diff
@@ -0,0 +1,179 @@
+            for root, dirs, files in os.walk(item_path):
+                # we only need the full source path to copy files to the destination,
+                # to create all necessary folders we also need the path suffix for
+                # the directory being walked and the target path based on the suffix
+                root_suffix = os.path.relpath(root, source_dir)
+                root_in_dest_dir = os.path.join(dest_dir, root_suffix)
+                # create directories
+                if not os.path.isdir(root_in_dest_dir):
+                    os.makedirs(root_in_dest_dir)
+                # copy files
+                for file in files:
+                    # create containing directories
+                    if not os.path.isdir(root_in_dest_dir):
+                        os.makedirs(root_in_dest_dir)
+                    shutil.copyfile(os.path.join(root, file), os.path.join(root_in_dest_dir, file))
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.parse()
+
+    # check the requested platform folder exists:
+    if not os.path.exists(parser.platform_dir_path):
+        print("platform folder path not found: {}".format(parser.platform_dir_path))
+        exit(1)
+
+    # make sure the dynamic folder exists and is empty
+    if os.path.exists(RUNTIME_FOLDER):
+        # cleanup the runtime folder
+        shutil.rmtree(RUNTIME_FOLDER)
+        # create new one
+        os.makedirs(RUNTIME_FOLDER)
+    else:
+        # create the dynamic folder
+        os.makedirs(RUNTIME_FOLDER)
+
+    # copy all shared fragments to the runtime folder
+    copy_directory_content(SHARED_FRAGMENTS_FOLDER, RUNTIME_FOLDER)
+
+    # copy contents of the current platform folder into it as well
+    copy_directory_content(parser.platform_dir_path, RUNTIME_FOLDER)
+
+    # copy any overrides
+    for override_folder in parser.override_folders:
+        if os.path.exists(override_folder):
+            print("adding ksappend override folder {}".format(override_folder))
+            copy_directory_content(override_folder, RUNTIME_FOLDER)
+        else:
+            print("requested ksappend override folder {} not found".format(override_folder))
+
+    # copy all the test files to runtime folder
+    available_test_files = []
+
+    # check the files actually exist
+    for test_path in parser.test_file_paths:
+        if os.path.exists(test_path) and os.path.isfile(test_path):
+            available_test_files.append(test_path)
+        else:
+            print("test file requested for ksappend substitution is missing: %s", test_path)
+
+    # copy existing files to the runtime folder
+    for test_path in available_test_files:
+        shutil.copy(test_path, RUNTIME_FOLDER)
+
+    # change working directory to the runtime folder or else
+    # Pykickstart will not find the fragments
+    os.chdir(RUNTIME_FOLDER)
+
+    # do the substitution
+    resolved_ks_files = []
+    for test_file in glob.glob("*.ks".format(os.sep)):
```

Comment:
Why the format is here?

Reply:
Indeed, this is not needed anymore and can be removed.